### PR TITLE
Fix datetime bug

### LINKED ISSUE #130: [PE-D][Tester D] Incorrect command result for datetime paramter
# Description

When passing `0000-01-01` into the datetime parameter for `list`, the command result shows that the date listed was `0001-01-01` instead.

## Steps to reproduce

Run the following command

```
list 0000-01-01
```

## Expected behaviour

The date for the command result should indicate `1st Jan 0000` instead of `1st Jan 0001`.

## Actual behaviour

`1st Jan 0001` is returned.

## Screenshots

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: marcuspang/ped#3

### PR DESCRIPTION
**Pull request by ruiqi7**
- [x] Invalidate datetimes with year 0000
- [x] Add test case
- [x] Update UG

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -120,12 +122,16 @@ Example:
 
 <div markdown="block" class="alert alert-info">
 
-**:information_source: Note:**<br>
+**:information_source: Notes about the parameters:**<br>
```
Comment (xav168) [inline docs/UserGuide.md]: would it be better to change it to
```suggestion
**:information_source: About the parameters:**<br>
```

File: docs/UserGuide.md [hunk 2]
```diff
@@ -120,12 +122,16 @@ Example:
...
 
 `MEAL_TYPE` is **not** case-sensitive.
 
 * E.g. `add n/hotpot c/500 t/dinner` is the same as `add n/hotpot c/500 t/dInNeR`.
 
+**:information_source: Notes about adding food items:**<br>
```
Comment (xav168) [inline docs/UserGuide.md]: would this be better?
```suggestion
**:information_source: Adding food items:**<br>
```

File: docs/UserGuide.md [hunk 3]
```diff
@@ -3,7 +3,8 @@ layout: page
 title: User Guide
 ---
 
-NutriGoals is a desktop app that tracks a user’s diet and calorie consumption. It is targeted at NUS students who wish to improve their current lifestyle by adopting a healthier one. Studies have shown the benefits of keeping 
+NutriGoals is a desktop app that tracks a user’s diet and calorie consumption. It is targeted at NUS students who wish 
+to improve their current lifestyle by adopting a healthier one. Studies have shown the benefits of keeping
```
Comment (nicolelim02) [inline docs/UserGuide.md]:
```suggestion
NutriGoals is a desktop app that tracks a user’s diet and calorie consumption. It is targeted at NUS students who wish to improve their current lifestyle by adopting a healthier one. Studies have shown the benefits of keeping 
```

File: docs/UserGuide.md [hunk 4]
```diff
@@ -63,9 +64,11 @@ as `nutrigoals.jar`. Do not delete or edit these files.
 * Food items displayed in Food List are sorted according to meal type, in the order: breakfast, lunch, dinner.
 * Upon launching the application, Food List will display food items recorded on the current day.
 * Progress Bar changes based on the total calorie intake for the current day as compared to the target calorie intake.
-  Refer to the `target` command [here](#setting-a-target-calorie-intake-target).
-* The percentage of Progress Bar can go beyond 100%, but note that if the target calorie intake is set to 0, the 
+* To check the total calorie intake for the current day and the target calorie intake, refer to the `review` command 
+[here](#viewing-a-summary-of-the-calorie-intake-progress-review).
```
Comment (nicolelim02) [inline docs/UserGuide.md]:
```suggestion
* To check the total calorie intake for the current day and the target calorie intake, refer to the `review` command [here](#viewing-a-summary-of-the-calorie-intake-progress-review).
```

File: docs/UserGuide.md [hunk 5]
```diff
@@ -236,9 +247,10 @@ Example:
 
 * `target 2500` sets a target calorie intake of 2500 calories for the current day and displays the list of foods for the current day.
 
-### Viewing a summary of the daily calorie intake: `review`
+### Viewing a summary of the calorie intake progress: `review`
 
-Shows the total calories consumed, the calorie target and the deficient or excess amount of calories for the day. The list of foods for the day will also be displayed.
+Shows the total calories consumed, the calorie target and the deficient or excess amount of calories for the current day. 
+The list of foods for the current day will also be displayed.
```
Comment (nicolelim02) [inline docs/UserGuide.md]:
```suggestion
Shows the total calories consumed, the calorie target and the deficient or excess amount of calories for the current day. The list of foods for the current day will also be displayed.
```

File: docs/UserGuide.md [hunk 6]
```diff
@@ -192,9 +202,10 @@ Example:
 
 **:information_source: Note:**<br>
 
-`DATE` must be a valid date for the command to be executed.
+`DATE` must be a valid date for the command to be executed. Note that entering `0000` as the year is considered invalid.
```
Comment (nicolelim02) [inline docs/UserGuide.md]: Would this be better?

```suggestion
`DATE` must be a valid date for the command to be executed. Note that the year `0000` is considered invalid.
```

**Review by nicolelim02 — APPROVED** [review]
Tested and it seems to be working!

**Pull request merged**